modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -67,20 +70,25 @@ public class TieredSpilloverCache<K, V> implements ICache<K, V> {
     /**
      * Maintains caching tiers in ascending order of cache latency.
      */
-    private final List<ICache<K, V>> cacheList;
+    private final Map<ICache<K, V>, Boolean> cacheList;
```

Comment:
> 1.lets define this type as LinkedHashMap explicitly as the cache will not work with any other map type since order will not be guaranteed.

My original plan was to do that. But I am creating a synchronized version of LinkedHashMap using `Collections.synchronizedMap` which only works with interface `Map`. Let me know if you have any other way to do this.
>2.cacheList to caches ?

Will change.

Reply:
>> 1.lets define this type as LinkedHashMap explicitly as the cache will not work with any other map type since order will not be guaranteed.

> My original plan was to do that. But I am creating a synchronized version of LinkedHashMap using Collections.synchronizedMap which only works with interface Map. Let me know if you have any other way to do this.

Sorry didn't get it. `LinkedHashMap` is of type Map so why there is issue with `Collections.synchronizedMap` ?

Reply:
Oh maybe I was not clear.
```Collections.synchronizedMap``` returns an interface `Map`, and if  `this.caches` is of concrete `LinkedHashMap`
 type, then compiler complains that it can't assign an interface `Map` to a concrete class( error: Required type expected `LinkedHashMap`, Provided `Map`) .

I can actually cast it using below. Should I fine I guess.
```
this.caches = (LinkedHashMap<ICache<K, V>, Boolean>) Collections.synchronizedMap(cacheListMap);
```